enhancements/metering/slim-install.md

```diff
@@ -0,0 +1,210 @@
+
+Currently, and by default, metering periodically imports and stores a small subset of the available Prometheus metrics in Hive long term, using the Prometheus query_range API. Historically, Presto did not have the ability to directly query Prometheus, but it does have the ability to integrate with Hive datasets. This results in the following workaround, where tables are needed to be first created in Hive, and then newer metrics would be inserted by Presto into existing tables in Hive using the `INSERT INTO ... VALUES(...)` operation. It's also worth noting that type of operation is particularly expensive for Presto in comparison to Postgres, which is designed as a more transactional-based system (OLTP).
+
+Due to the nature of periodic importing, most of the components are sitting idle, especially in the case of Presto which is still utilizing resources despite no queries being processed or evaluated.
+
+With the introduction of Postgres, and the removal of Hive and Presto components, we can continue to utilize the existing Prometheus metrics importer implementation, adapting to the native Postgres data types, and the result is a more streamlined approach to data flow and an overall less complex stack.
+
+This implementation would allow metering to continue using the core, existing CustomResourceDefinitions (CRDs) already exposed in previous releases:
+
+- `ReportDataSource`: exposes an existing table in Postgresql that can be used for reporting needs.
+- `ReportQuery`: defines a SQL query or view that references a `ReportDataSource` table, or a table created by a `Report` object.
+- `Report`: evaluates a templated SQL query defined in an existing `ReportQuery` custom resource and stores those query results in a Hive (now Postgres) table.
+
+At a high-level, the following must be done:
+
+- Create a `PostgresTable` CRD, which is an abstraction that represents an individual table in Postgres.
+- Remove (i.e. avoid reconciling) the internal `HiveTable` and `PrestoTable` custom resources.
+- Update the existing `ReportDataSource` promQL queries to reference the recording rules defined in the metering rule group.
+- Update the existing `ReportQuery` SQL queries and views to index Prometheus labels stored as `jsonb` objects in Postgres tables.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Interacting with Postgres data types
+
+In order to conform with native Postgres data types, any Prometheus labels returned by the query_range API endpoint will need to be represented as the `jsonb` type. This could result in the following `PostgresTable` custom resource instantiation, where metric data is partitioned by datetime and the resultant table column name and types are explicitly defined:
+
+```yaml
+apiVersion: metering.openshift.io/v1
+kind: PostgresTable
+metadata:
+  name: reportdatasource-metering-node-allocatable-cpu-cores
+spec:
+  columns:
+  - name: amount
+    type: float8
+  - name: timestamp
+    type: timestamptz
```

Comment:
This might go up in the details section or something: I think Presto has a slightly different way of dealing with time zone. Need to research a bit.

Reply:
I looked at this some more. I think we always have timestamp with timezone type timestamp data in Metering as Presto supports that. I also think it's always UTC as that is Prometheus' export standard. I see one place in the code where we strip the timezone value because Hive doesn't support the type.